cssselect/parser.py

```diff
@@ -564,6 +591,24 @@ def parse_arguments(stream):
                 "Expected an argument, got %s" % (next,))
 
 
+def parse_relative_selector(stream):
+    arguments = []
+    stream.skip_whitespace()
+    next = stream.next()
+    if next in [('DELIM', '+'), ('DELIM', '-'), ('DELIM', '>'), ('DELIM', '~')]:
+        arguments.append(next)
```

Comment:
This is working _exclusively_ for selectors that begin with "+", ">", etc. but `parse("foo:has(bar)")` results in `[Selector[Relation[Element[foo]:has([])]]]` (nothing inside the "has").
That might be the intention for now, feel free to ignore this suggestion if that's the case :smile:  

```suggestion
    if next in [('DELIM', '+'), ('DELIM', '-'), ('DELIM', '>'), ('DELIM', '~')]:
        arguments.append(next)
    elif next.type in ('IDENT', 'STRING', 'NUMBER'):
        arguments.append(Element(element=next.value))
```

Reply:
the reasoning for this solution was that [here](https://www.w3.org/TR/selectors-4/#relational) it is said that `has` expects a  relative selector which starts with a combinator
`<relative-selector> = <combinator>? <complex-selector>`
but I see your suggestion is just a couple of lines, so no use in not including it